addons/stock/tests/test_product.py

```diff
@@ -276,3 +276,103 @@ def test_search_product_template(self):
         res = self.env['product.template'].name_search(name='123', operator='not ilike')
         res_ids = [r[0] for r in res]
         self.assertNotIn(template.id, res_ids)
+
+    def test_qty_available_multi_warehouse_location_context(self):
+        """
+        Test computed quantities of a product with multiple warehouses and locations in context
+        """
+        warehouse1 = self.env["stock.warehouse"].create({"name": "Primary Warehouse", "code": "PWH",
+                                                         "delivery_steps": "pick_ship"})
+        wh1_location = warehouse1.lot_stock_id
+        wh1_sublocation = self.env["stock.location"].create({"name": "Sublocation", "location_id": wh1_location.id})
+
+        warehouse2 = self.env["stock.warehouse"].create({"name": "Secondary Warehouse", "code": "SWH"})
+        wh2_location = warehouse2.lot_stock_id
+        wh2_sublocation = self.env["stock.location"].create({"name": "Sublocation", "location_id": wh2_location.id})
+
+        self.env['stock.quant'].create({
+            'product_id': self.product_3.id,
+            'location_id': wh1_location.id,
+            'quantity': 7.0})
+        self.env['stock.quant'].create({
+            'product_id': self.product_3.id,
+            'location_id': wh1_sublocation.id,
+            'quantity': 2.0})
+
+        self.env['stock.quant'].create({
+            'product_id': self.product_3.id,
+            'location_id': warehouse2.lot_stock_id.id,
+            'quantity': 5.0})
+        self.env['stock.quant'].create({
+            'product_id': self.product_3.id,
+            'location_id': wh2_sublocation.id,
+            'quantity': 10.0})
+
+        # Check if quantity at wh1 and wh1_loc is correct
+        self.assertEqual(self.product_3.with_context(warehouse=warehouse1.id,
+                                                     location=wh1_location.id).qty_available, 9)
```

Comment:
```suggestion
        self.assertEqual(self.product_3.with_context(warehouse=warehouse1.id,location=wh1_location.id).qty_available, 9)
```
Can you also reformat all other `assert` ?